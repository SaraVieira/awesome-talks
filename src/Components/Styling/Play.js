import styled from 'styled-components'

export default styled.button`
    background: #282828;
    border-radius: 50% / 10%;
    color: ${props => props.theme.primary};
    font-size: 1em;
    height: 3em;
    padding: 0;
    text-align: center;
    text-indent: 0.1em;
    transition: all 150ms ease-out;
    width: 4em;
    position: absolute !important;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    border: none;
    opacity: 0.8;
    cursor: pointer;

    &:hover {
        background: #ff0000;
    }

    &:before {
        background: inherit;
        border-radius: 5% / 50%;
        bottom: 9%;
        content: '';
        left: -5%;
        position: absolute;
        right: -5%;
        top: 9%;
    }

    &:after {
        border-style: solid;
        border-width: 1em 0 1em 1.732em;
        border-color: transparent transparent transparent
            rgba(255, 255, 255, 0.75);
        content: ' ';
        font-size: 0.75em;
        height: 0;
        margin: -1em 0 0 -0.75em;
        top: 50%;
        position: absolute;
        width: 0;
    }
`
