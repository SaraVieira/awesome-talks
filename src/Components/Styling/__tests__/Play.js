import React from 'react'
import Shallow from '../../../Tests/shallow'
import Play from '../Play'

test('Renders Play', () => {
    const Wrapper = Shallow(<Play />)

    expect(Wrapper).toHaveStyleRule('background', '#282828')
    expect(Wrapper).toHaveStyleRule('height', '3em')
    expect(Wrapper).toMatchSnapshot()
})
