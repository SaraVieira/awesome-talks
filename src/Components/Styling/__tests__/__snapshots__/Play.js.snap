// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Play 1`] = `
.c0 {
  background: #282828;
  border-radius: 50% / 10%;
  font-size: 1em;
  height: 3em;
  padding: 0;
  text-align: center;
  text-indent: 0.1em;
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  width: 4em;
  position: absolute !important;
  top: 50%;
  left: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
  border: none;
  opacity: 0.8;
  cursor: pointer;
}

.c0:hover {
  background: #ff0000;
}

.c0:before {
  background: inherit;
  border-radius: 5% / 50%;
  bottom: 9%;
  content: '';
  left: -5%;
  position: absolute;
  right: -5%;
  top: 9%;
}

.c0:after {
  border-style: solid;
  border-width: 1em 0 1em 1.732em;
  border-color: transparent transparent transparent rgba(255,255,255,0.75);
  content: ' ';
  font-size: 0.75em;
  height: 0;
  margin: -1em 0 0 -0.75em;
  top: 50%;
  position: absolute;
  width: 0;
}

<button
  className="c0"
/>
`;
